utils/das_data_sorter.py

```diff
@@ -1,76 +1,170 @@
+            if number == 0 or pd.isna(number):
+                # None and zeroes are ignored
+                length -= 1
+                continue
+            elif number > 0 or number < 0:
+                # 'number' value is valid
+                total += number
+            else:
+                # 'number' value is invalid
+                raise ValueError("Data point invalid and is neither zero nor None. The data point was "+str(number)+". ("+str(type(number))+")")
+        
+        try:
+            return total/length
+        except ZeroDivisionError:
+            # in the event where length = 0, due to all the elements in data_array being ignored
+            return 0 
+
+    def average_data(self, data:D) -> D:
+        '''Returns a column of new data points that has been averaged, based on specified time unit. 
+
+        Finds the average of all data points within the same time interval. Appends this new value into an array, then continues
+        to the next time interval. Once all time intervals have been accounted for, returns the set of new data points.
+        '''
+        avg_data = []
+
+        for index_array in self.indexes:
+            avg = self.mean(data[index_array])
+            avg = round(avg, ndigits=2)
+            avg_data.append(avg)
+
+        return avg_data
+
+    def location_data(self, data:D) -> D:
+        '''Handles the location data and returns the data point at the first element of each time interval. 
+
+        gps_location requires a separate function, because the values are strings, whereas the other data columns are 
+        integers/floats. Handles it so that it will always return the location data at the first element of each time interval.
+        '''
+        new_location_data = []
+        for index_array in self.indexes:
+            first_element = index_array[0] 
+            new_location = data[first_element]
+            new_location_data.append(new_location)
+        
+        return new_location_data
+    
+    def print_output_file(self) -> None:
+        '''Creates new CSV file and writes new data onto CSV file.'''
+        final = pd.DataFrame({
+            "time": self.time,
+            "gps": self.gps,
+            "gps_location": self.gps_location,
+            "gps_course": self.gps_course,
+            "gps_speed": self.gps_speed,
+            "gps_satellites": self.gps_satellites,
+            "aX": self.ax,
+            "aY": self.ay,
+            "aZ": self.az,
+            "gX": self.gx,
+            "gY": self.gy,
+            "gZ": self.gz,
+            "thermoC": self.thermoc,
+            "tempF": self.thermof,
+            "pot": self.pot,
+            "cadence": self.cadence,
+            "power": self.power,
+            "reed_velocity": self.reed_velocity,
+            "reed_distance": self.reed_distance
+        })
+        final.to_csv(args.output, index=False)
+        print("Success! Check in the same folder for the new CSV file.")
```

Comment:
The output file isn't located in the same folder as the input, it's located wherever the user specified. Maybe instead print `Success! Output written to {args.output}`

Reply:
I was about to say this!